crates/rpc/rpc/src/eth/api/changes.rs

```diff
@@ -0,0 +1,76 @@
+use crate::eth::error::{EthApiError, EthResult};
+use crate::EthApi;
+use reth_primitives::{Account, Address, BlockId, U256};
+use reth_provider::{AccountExtReader, BlockReaderIdExt, StateProviderFactory};
+use revm_primitives::HashMap;
+
+impl<Provider, Pool, Network> EthApi<Provider, Pool, Network>
+where
+    Provider: AccountExtReader + BlockReaderIdExt + StateProviderFactory + 'static,
+{
+    pub(crate) fn account_changes_in_block(
+        &self,
+        block_id: impl Into<BlockId>,
+    ) -> EthResult<Option<HashMap<Address, Account>>> {
```

Comment:
We could return just a `HashMap`, Option::None and HashMap::is_empty are same states.

Reply:
Thanks, I'm relatively new to rust so am still getting used to what allocs and doesn't.  I had noticed that we often use `EthResult<Option<Vec<..>>>` so was following that idiom.

Reply:
@mattsse what does it mean for RPC if it is `Option`?

Reply:
I thought about this, and assume `Option<HashMap>` would allow the JSONRPC `result` to be `null` vs. `{}`, I personally think `{}` makes more sense here so removed the `Option` (more details here: https://github.com/paradigmxyz/reth/pull/3768#discussion_r1264558692).

But definitely let me know if my understanding of the difference between `None` and `HashMap::default()` is wrong!